### PR TITLE
ArC: fix some corner cases in Build Compatible Extensions

### PR DESCRIPTION
**Pull request by Ladicek**
- support observing interceptors in `@Registration`
- fix reporting priority of beans that are not alternatives (not very useful in standard CDI, but in ArC, we go beyond the standard)
- fix NPE in `ClassInfo.methods()` (and `fields()`) for `java.lang.Object`
- fix (and simplify) invoking extension methods

### CODE REVIEW

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/cdi/bcextensions/RegistrationTest.java [hunk 1]
```diff
@@ -57,6 +68,14 @@ public void observers(ObserverInfo observer, Types types) {
                 observerQualifierCounter.addAndGet(observer.qualifiers().size());
             }
         }
+
+        @Registration(types = MyInterceptor.class)
+        public void interceptors(BeanInfo interceptor, Messages msg) {
```
Comment (manovotn) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/cdi/bcextensions/RegistrationTest.java:73]: Do we have a test case where the param is `InterceptorInfo` instead?

**Review by manovotn — APPROVED** [review]

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/cdi/bcextensions/RegistrationTest.java (hunk 1 above)
Comment (Ladicek) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/cdi/bcextensions/RegistrationTest.java:73]: I don't think we do, no.

**Review by mkouba — APPROVED** [review]
Loos good.

**Comment by Ladicek** [comment]
`HibernateNamedPersistenceUnitTest` is flaky, gonna merge.

**Pull request merged**